rollup/internal/controller/sender/sender.go

```diff
@@ -232,6 +232,12 @@ func (s *Sender) SendTransaction(contextID string, target *common.Address, data
 	}
 
 	if err := s.client.SendTransaction(s.ctx, signedTx); err != nil {
+		// Delete the transaction from the pending transaction table if it fails to send.
+		if updateErr := s.pendingTransactionOrm.DeletePendingTransactionByTxHash(s.ctx, signedTx.Hash()); updateErr != nil {
```

Comment:
Why not move `s.pendingTransactionOrm.InsertPendingTransaction` after `s.client.SendTransaction` inside `SendTransaction`?

Also, what type of error do you have in mind here?

Reply:
> Also, what type of error do you have in mind here?

answered in another comment.

> move s.pendingTransactionOrm.InsertPendingTransaction after s.client.SendTransaction

yes. it's the previous implementation, while it cannot prevent forceful shutdown. but during graceful shutdown it does not have any problems (due to golang's context), so this error never happens in sepolia and mainnet before.